src/shogun/distance/Distance.h

```diff
@@ -354,6 +354,15 @@ class CDistance : public CSGObject
 		/// matrix precomputation
 		void do_precompute_matrix();
 
+		/**
+		 * Checks the compatibility between two supplied features
+		 *
+		 * @param l left hand side features
+		 * @param r right hand side features
+		 * @return true if the features are compatible
+		 */
+		virtual bool check_compatibility(CFeatures* l, CFeatures* r);
```

Comment:
maybe this one should print a warning if things dont work. And then the SG_ERROR is thrown afterwards